### DISCUSSION TITLE
How to differentiate attribute paths?

### DISCUSSION THREAD

**Discussion by Baptistemontan** [TedDriggs/darling#302] [Q&A] [answered]
In this example:
```rust
#[derive(Debug, FromField)]
#[darling(attributes(table, i18n))]
pub(crate) struct TableRowField {
    pub(crate) ident: Option<syn::Ident>,
    pub(crate) ty: syn::Type,

    #[darling(default)]
    pub(crate) skip: bool,
}
```
How do I know if `skip` comes from the `table` attr or the `i18n` attr ?

**Comment by TedDriggs** [comment, marked as answer]
There isn't a native way to do so; `darling` internally generates a block that essentially says, "If the attribute ident is one of the ones I'm supposed to parse, parse its inner contents" and doesn't retain the outer attribute ident.

What you can probably do instead is this...

```rust
#[derive(Debug, FromAttributes)]
struct FieldI18nAttrs {
    // ... fields
}

fn parse_i18n_attrs(attrs: Vec<Attribute>) -> darling::Result<FieldI18nAttrs> {
    FieldI18nAttrs::from_attributes(&attrs)
}

#[derive(Debug, FromField)]
#[darling(attributes(table), forward_attrs(i18n))]
struct TableRowField {
    #[darling(with = parse_i18n_attrs)]
     attrs: FieldI18nAttrs,

     ident: Option<syn::Ident>,
     ty: syn::Type,

    #[darling(default)]
    skip: bool,
}
```

This approach takes advantage of the `attrs` field supporting `#[darling(with = ...)]` to forward the secondary items to the `attrs` field, where they can then be parsed on their own.

**Reply by Baptistemontan** [reply]
I'll try that, thanks for the example !
